src/EVM/Solvers.hs

```diff
@@ -128,7 +136,9 @@ withSolvers solver count timeout cont = do
                 "unknown" -> pure Unknown
                 "sat" -> if null refineEqs then Sat <$> getModel inst cexvars
                          else do
-                              _ <- sendScript inst (SMT2 refineEqs mempty mempty)
+                              let refinedSMT2 = SMT2 refineEqs mempty mempty (ps <> refps)
+                              writeSMT2File refinedSMT2 fileCounter "refined"
```

Comment:
what do you think about doing this inside of `sendLine` and friends? and just append every line we send for each `SMT2` to the same file?

Reply:
Can we do that as a follow-up? You mean that all `send...` functions should now take another parameter, an open IO (`Maybe FileIO` or something?) for a file that is open? Honestly, I am a bit scared of the file IO in Haskell, I can barely use it. I have a feeling I'd make a mess of it and then this PR will stall even longer :(

Reply:
Anyway, I'm gonna do it, but I'm sure it's just gonna be wrong. But anyway, let me see.

Reply:
Well, I had a go, but it's wrong, and quite a bit more code. It misses the Props being there, though it's possible to add it, but anyway. The sendLine sends `(reset)`-s and timeouts, so that will need to be not sent. Not sure this is cleaner than a 2-liner.

Reply:
I just reverted it, it's here cf769b95faf77947ac36ebcba367b6b7a05e8147 but it's incomplete and a lot more noisy than the 2 lines of `writeSMT2File`. It is also has a bunch of complications and I'm not sure it's worth it, I think those 2 lines are easy enough.